### PR TITLE
Audio/Video Title and Artist Metadata

### PR DESCRIPTION
**Pull request by the8472**
Depends on bakape/thumbnailer/pull/3

### CODE REVIEW

File: client/posts/render/image.ts [hunk 1]
```diff
@@ -70,42 +70,30 @@ function renderFigcaption(post: Element, data: ImageData, reveal: boolean) {
...
+    info.querySelector(".media-title").textContent = data.title
+    info.querySelector(".media-artist").textContent = data.artist
+    info.querySelector(".has-audio").hidden = !data.audio
+    if (data.length) {
+        info.querySelector(".media-length").textContent = readableLength(data.length)
+    }
+    info.querySelector(".is-apng").hidden = !data.apng
+    info.querySelector(".filesize").textContent = readableFilesize(data.size)
+    info.querySelector(".dims").textContent = `${data.dims[0]}x${data.dims[1]}`
+
```
Comment (the8472) [inline client/posts/render/image.ts]: > Not all of these elements are always present.

Under which circumstances does the figcaption need to be updated? Either there's no image or the image is in its final state, no?

Comment (bakape) [inline client/posts/render/image.ts]: In hidden thumbnail mode. I guess rendering the whole thing is not strictly necessary, but there are too many things, that rerender an image and/or figcaption, and several flag combinations for each. To simplify this and avoid a lot of extra bookkeeping, I rerender the figure and figcaption on each change. Can be optimised away, of course.

**Comment by the8472** [comment]
Better?

**Comment by bakape** [comment]
Yeah, just perfect.

**Pull request merged**